Services/UI/OJS.Services.Ui.Data/Implementations/ParticipantsDataService.cs

```diff
@@ -25,17 +25,30 @@ public ParticipantsDataService(OjsDbContext db)
             => this.GetAllByContestByUserAndIsOfficial(contestId, userId, isOfficial)
                 .FirstOrDefaultAsync();
 
-        public Task<Participant?> GetWithContestByContestByUserAndIsOfficial(int contestId, string userId, bool isOfficial)
+        public Task<Participant?> GetWithContestAndSubmissionDetailsByContestByUserAndIsOfficial(int contestId, string userId, bool isOfficial)
             => this.GetAllByContestByUserAndIsOfficial(contestId, userId, isOfficial)
                 .Include(p => p.Contest)
-                .ThenInclude(c => c.ProblemGroups)
-                .ThenInclude(pg => pg.Problems)
+                    .ThenInclude(c => c.ProblemGroups)
+                        .ThenInclude(pg => pg.Problems)
+                            .ThenInclude(p => p.SubmissionTypesInProblems)
+                                .ThenInclude(sp => sp.SubmissionType)
+                .Include(p => p.Submissions)
+                .Include(p => p.Scores)
+                .Include(p => p.ProblemsForParticipants)
                 .FirstOrDefaultAsync();
 
         public IQueryable<Participant> GetAllByUser(string? userId)
             => this.DbSet
                 .Where(p => p.UserId == userId);
 
+        public IQueryable<Participant> GetAllWithContestSubmissionsAndProblemsByUser(string? userId)
+            => this.DbSet
+                .Where(p => p.UserId == userId)
+                .Include(p => p.Submissions)
```

Comment:
Can this be avoided? Submissions can potentially be very big set with heavy data and here you get all the submissions of a participant. You can return IEnumerable with service model and use only the properties you need.

Reply:
Method has no usages so those includes were not needed. Should I remove these includes? 

![image](https://github.com/SoftUni-Internal/OpenJudgeSystem/assets/100528439/5edf6482-24d3-4baf-93ba-4c5d62db5496)

Reply:
Go ahead and delete unused methods which you have modified

Reply:
All modified methods have been changed. Unfortunately there was a bug in Rider and while double checking this method in particular I found there are in fact usages, so I am refactoring it to use service models.